contracts/instance/IBaseStore.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.20;
+
+import {Blocknumber} from "../type/Blocknumber.sol";
+import {Key32, KeyId} from "../type/Key32.sol";
+import {NftId} from "../type/NftId.sol";
+import {ObjectType} from "../type/ObjectType.sol";
+import {StateId} from "../type/StateId.sol";
+
+import {ILifecycle} from "../shared/ILifecycle.sol";
+
+interface IBaseStore is ILifecycle {
+
+    error ErrorBaseStoreTypeUndefined(ObjectType objectType);
+    error ErrorBaseStoreAlreadyCreated(Key32 key, ObjectType objectType);
+    error ErrorBaseStoreNoLifecycle(ObjectType objectType);
+    error ErrorBaseStoreStateZero(Key32 key);
+    error ErrorBaseStoreNotExisting(Key32 key);
+
+    event LogBaseStoreInfoCreated(ObjectType objectType, KeyId keyId, StateId state, address createdBy, address txOrigin);
```

Comment:
not so sure anymore that we need logging here:
- complete and more business oriented logging in product store seems to be more meaningful
- on the other side: not having low level logging here means no logging at all should logging be forgottenin product store